src/lib/helpers/deepMerge.js

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************************
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ ****************************************************************************************/
+
+'use strict';
+
+var isPlainObject = require('../helpers/isPlainObject');
+
+var clone = function (value) {
+  if (isPlainObject(value)) {
+    return deepMerge({}, value);
+  }
+
+  if (Array.isArray(value)) {
+    return deepMerge([], value);
+  }
+
+  return value;
+};
+
+function deepMerge(target) {
+  var sources = Array.prototype.slice.call(arguments, 1);
+  return sources.reduce(function (merged, source) {
+    if (source == null) {
+      return merged;
+    }
+
+    Object.keys(source).forEach(function (key) {
+      var mergedValue = merged[key];
+      var sourceValue = source[key];
+
+      if (sourceValue === undefined) {
+        return;
```

Comment:
I think I understand why you wouldn't want undefined to overwrite previous values. Namely, if you had a data element that returns undefined, it shouldn't overwrite previous values. However, Is there a way to remove values from a previous object?

Like `deepMerge({a: 1, b: 2}, {a: __something__}) === {b: 2}`

Reply:
Yeah, what you describe is accurate. Notably, some packages, like lodash's merge utility, exhibit the same behavior as what I've provided here, whereas other libraries, like `deepmerge`, overwrite the previous value with `undefined`. I suppose which approach is taken is subjective, but particularly considering the use cases we're most aware of, I think not overwriting values with `undefined` probably makes the most sense.

There is no way to remove values that were merged from prior objects.